joomla/joomla-cms
# [4.0] Fix #21357 when multilanguage is implemented

Code error in #21357

### Summary of Changes
Making sure we are in multilang before using forcedLanguage.
Changing the code to get the article language as `$this->element['language']` is wrong in this case.


### Testing Instructions
Create a multingual site with articles set to some languages and other set to All languages.
When using the new "New Menu Item" button, the modal presented should take into account the language of the article for the Menu item created.
For an article tagged to ALL languages the language for the menu item can be anything (no forcedLanguage).
But for an article tagged to a specific language, the menu item should also be tagged to the same language.
This is unrelated to the Menu, which can be set to ALL or any language.

### Before patch
We can choose in the modal any language for the menu item.

### After patch
When the article is not tagged to ALL, the new menu item is already tagged to the same language as the article and it can't be changed.
<img width="1373" alt="screen shot 2018-08-03 at 17 15 42" src="https://user-images.githubusercontent.com/869724/43650798-02dd9b08-9741-11e8-9ded-3e77f680bcfc.png">

### Documentation Changes Required

yes when doc is made for #21357

@bene-we
